### ISSUE TITLE
Auto resolve to new IP address

### ISSUE THREAD

**Issue by alandotic** [sebastianhaberey/nsd#32]
### Description

After the service registration, i changed my ip address in setting and back to app, the discovery result still show the old ip address, i tried to unregister and register the service and it still resolve to old IP address.

I closed the app and reopen, it still resolve to old IP address.

i killed the app and reopen, then it resolve to new IP address.

 

### Scenario

Use the plugin in my own application via Flutter dependency

### Platform

Android

**Comment by sebastianhaberey** [comment]
Hi @alandotic,

I think I know why this is happening, but I don't have a solution yet. I'll keep you updated.

**Comment by sebastianhaberey** [comment]
I've had a closer look at this now. 

The IP address is looked up _once_ by the nsd plugin - at the time when the service is detected. This is more of a gimmick because it was a popular request by users. The problem is, when the IP address changes afterwards, the plugin will not be aware of it. This is because the DNS-SD protocol itself and the corresponding APIs on Android / iOS / macOS / Windows do not actually _know_ IP addresses. Instead, they rely on host names. So if a service's host name changes, the change will propagate through the system. But changes in IP addresses will not.

So basically the IP address you get with the service is valid at the time of service detection. I don't know your use case, but there's a few things you can do:

- Use host names. You should always be able address a service by the supplied host name. This is the preferred way.
- You could do the IP lookup yourself, shortly before you actually use it. This is really easy in dart:

```dart
import 'dart:io';

final addresses = await InternetAddress.lookup(host);
```

Does this help in your case?
